web/auth/confirm.go

```diff
@@ -127,32 +127,41 @@ func ConfirmSuccess(c echo.Context, inst *instance.Instance, state string) error
 	return c.Redirect(http.StatusSeeOther, redirect.String())
 }
 
-func checkRedirectToManager(c echo.Context) (*url.URL, error) {
+func checkRedirectToAuthorized(c echo.Context) (*url.URL, error) {
 	inst := middlewares.GetInstance(c)
+	redirect := c.FormValue("redirect")
+	u, err := url.Parse(redirect)
+	if err != nil {
+		return nil, echo.NewHTTPError(http.StatusBadRequest, "bad url: could not parse")
+	}
+
+	if ok := checkRedirectToManager(inst, u); ok {
+		return u, nil
+	}
+
+	for _, host := range config.GetConfig().AuthorizedForConfirm {
+		if host == u.Host {
```

Comment:
I wonder if this should be Host only or Host + Port (like in https://tool.domain.example:8080)

Reply:
It can be both. https://pkg.go.dev/net/url#URL says:

> Host        string    // host or host:port (see Hostname and Port methods)